azure-pipelines.yml

```diff
@@ -15,16 +15,139 @@ variables:
 - group: DevOps Access Keys
 - name: version
   value: $(grep -i 'set version' meta.yaml | awk '{ print $5}' | sed 's/"//g')
+- name: TagName
+  value: "TestTag-$((version))"
 
 stages:
 
 #####################################
 # Linux Stages - Microsoft Hosted VMs
 #####################################
 
+### Dependency preparation stage
+- stage: linux_prep_deps
+  displayName: 'Linux Prepare Dependencies'
+  jobs:
+  - job: RunDeps
+    displayName: 'Runtime Dependencies'
+    timeoutInMinutes: 20
+    pool:
+      vmImage: 'ubuntu-16.04'
+    variables:
+    - name: VERSION
+      value: $(grep -i 'set version' meta.yaml | awk '{ print $5}' | sed 's/"//g')
+    steps:
+
+    # Manually adding conda to the PATH is a step required by Microsoft
+    - bash: echo "##vso[task.prependpath]$CONDA/bin"
+      displayName: 'Add conda to PATH'
+
+    - bash: |
+        echo "##vso[task.setvariable variable=VERSION]$(VERSION)"
+      displayName: 'Expose Version string as environment variable.'
+
+    # Set up environment and run the build
+    - bash: conda install --yes --quiet -c conda-forge -c fermi python=2.7 anaconda-client conda-build libiconv pathlib2 fermi-repoman
+      displayName: 'Set up build environment'
+
+    - bash: |
+        mkdir components && cd components
+        repoman --remote-base https://github.com/fermi-lat checkout-list ../packageList.txt
+      displayName: 'Download Sources'
+
+    - bash: |
+        echo $VERSION
+        conda build --python=2.7 -c conda-forge -c fermi components/Fermitools-dependencies
+      displayName: 'Conda-Build Runtime Dependencies'
+
+    - bash: anaconda -v -t $(Jasercion-Anaconda-Api) upload -u fermi --label main --label dev --label $(TagName) --force /usr/share/miniconda/conda-bld/linux-64/fermitools-dependencies-*
+      displayName: 'Upload Runtime Dependencies'
+
+  - job: BuildDeps
+    displayName: 'Buildtime Dependencies'
+    timeoutInMinutes: 20
+    pool:
+      vmImage: 'ubuntu-16.04'
```

Comment:
At some point we may want to set these image names as variables that are hardcoded at the top of the document just for ease of update